### ISSUE TITLE
http: serving "abc/" as "abc/index.html"

### ISSUE THREAD

**Issue by jcw** [eclipse/mosquitto#706]
Mosquitto could be the single-tool solution for (quasi-) serverless applications, i.e. rich JavaScript applications (React, Angular, etc) which load from a 100% static web server, and then do all their dynamic work through WebSockets and MQTT. I think there's just one _tiny_ enhancement missing.

Right now, if I understand correctly, Mosquito can serve through a websocket, and it can server static files over http. Unfortunately, the default mode for React apps (and tools such as Gatsby) is to use sanitised URLS, i.e. "http://.../abc/" instead of "http://.../abc/index.html" or ""http://.../abc.html".

This leads to errors here: https://github.com/eclipse/mosquitto/blob/master/src/websockets.c#L519

Would it be possible to add logic to look for the `index.html` file if a directory path is encountered? I can probably create a pull request for such an enhancement, but my question is really: would there be any interest to add something like this to the official repo, if the change is reviewed and found to properly address this issue?

**Comment by ralight** [comment]
Yes, that sounds like a sensible suggestion, thanks for bringing it up.

**Comment by ralight** [comment]
I've just pushed a commit to the develop branch that should fix this, would you be able to test it out?
